### ISSUE TITLE
License expiration error

### ISSUE THREAD

**Issue by xipengshen** [CoCoPIE-Group/XGen-Report#19]
# Problem
The license of XGen is valid as the start of XGen shows "Your license has 90 days left". But at the end of compatibility test, the following error appeared: 
```
ERROR: test on device "RF8MA1SEARB" failed
The license for XGen inference engine has expired.
```

# Setup
I was remotely logged into eb2-3226-lin07 and ran XGen on that remote machine. But the smartphone was connected to my local Mac laptop. 

# Log
[xshen5@eb2-3226-lin07:> run_xgen
Existing XGen containers found.
Running multiple instances could result in file damage if multiple users share one XGen Data or Projects directory.
Successfully entering XGen environment. Next, you can type command "XGen" to interact with the powerful toolchain.
(xgen) root@eb2-3226-lin07:> XGen

XGen CLI by CoCoPIE
XGen(v1.0.17)

A DNN optimizer built on compression-compilation co-design.
It makes AI deployment sweet as pie by simultaneously
optimizing model size, accuracy, speed, and more.

© 2022 CoCoPIE Inc. All Rights Reserved.

Third Party Software License Disclaimer In:
 /root/third_party_license.txt

Your license has 90 days left.

Please choose your device(s):
> [*] RF8MA1SEARB model:SM_G973U1                                                                                                                                                                  
Press <space> or <tab> for multi-selection, <up> and <down> or letter key <j> and <k> to move, <enter> to accept.                                                                                  

Please choose the base model (and the default dataset) to start with:
  Image classification I: EfficientNet (ImageNet)                                                                                                                                                  
  Image classification II: ResNet (ImageNet)                                                                                                                                                       
> Image classification III: ResNet (Cifar)                                                                                                                                                         
  Image classification IV: MobileNet (ImageNet)                                                                                                                                                    
  Object detection: Yolov5 (CoCo)                                                                                                                                                                  
  Segmentation: UNet (ISBI-2012)                                                                                                                                                                   
  Video classification: R2+1d (UCF101)                                                                                                                                                             
  Video super resolution: WDSR (DIV2K)                                                                                                                                                             
  Your own model                                                                                                                                                                                   

Do you choose to download common AI model(ResNet) now?
> Yes                                                                                                                                                                                              
  No                                                                                                                                                                                               

Will you use the default dataset: Cifar?
> Yes                                                                                                                                                                                              
  No                                                                                                                                                                                               

What do you want to do? (Pruning recommended): 
> Compatibility test                                                                                                                                                                               
  Pruning                                                                                                                                                                                          
  Scaling                                                                                                                                                                                          
  Customized operation                                                                                                                                                                             

Which GPU(s) do you want to use: 
  [ ] GPU 0: Quadro K600 (UUID: GPU-e08cba0c-cb67-dd6b-54ec-adaf92784537)                                                                                                                          
> [*] GPU 1: NVIDIA GeForce GTX TITAN X (UUID: GPU-cc712512-b07f-d448-0baf-293e495924ec)                                                                                                           
Press <space> or <tab> for multi-selection, <up> and <down> or letter key <j> and <k> to move, <enter> to accept.                                                                                  

What is the batch size per GPU: 128
What is the learning rate: 0.01
How many epochs: 200
You chose "Compatibility test". We will create a temporary workplace for you, and will delete it after the testing.
2022-10-01 03:23:05,142 - root - INFO - AIMET
Your current workplace is /tmp/8b7f7da48acd4ed18a60d8d4ab3744c5
A new search is started!
****************************config summary************************************************
xgen-config-path:  /tmp/8b7f7da48acd4ed18a60d8d4ab3744c5/xgen_config.json
xgen-workplace:  /tmp/8b7f7da48acd4ed18a60d8d4ab3744c5
xgen-resume:  False
xgen-mode:  compatible_testing
xgen-pretrained-model-path:  ./checkpoint/ckpt.pth
Detail args:  {'origin': {'common_train_epochs': 200, 'root_path': './___workplace___/', 'pretrain_model_weights_path': None, 'train_data_path': './data', 'train_label_path': None, 
'eval_data_path': './data', 'eval_label_path': None, 'scaling_factor': 2, 'num_classes': 10, 'batch_size': 128, 'learning_rate': 0.01}, 'general': {'user_id': 'test', 'work_place': 
'/tmp/8b7f7da48acd4ed18a60d8d4ab3744c5', 'random_seed': 3407, 'enable_ddp': False, 'CUDA_VISIBLE_DEVICES': '1', 'tran_scripts_path': None}, 'prune': {'sp_store_weights': None, 'sp_lars': False, 
'sp_lars_trust_coef': 0.001, 'sp_backbone': False, 'sp_retrain': False, 'sp_admm': False, 'sp_admm_multi': False, 'sp_retrain_multi': False, 'sp_config_file': None, 'sp_subset_progressive': 
False, 'sp_admm_fixed_params': False, 'sp_no_harden': False, 'nv_sparse': False, 'sp_load_prune_params': None, 'sp_store_prune_params': None, 'generate_rand_seq_gap_yaml': False, 
'sp_admm_update_epoch': 5, 'sp_admm_update_batch': None, 'sp_admm_rho': 0.001, 'sparsity_type': 'block_punched', 'sp_admm_lr': 0.01, 'admm_debug': False, 'sp_global_weight_sparsity': False, 
'sp_prune_threshold': -1.0, 'sp_block_irregular_sparsity': '(0,0)', 'sp_block_permute_multiplier': 2, 'sp_admm_block': '(8,4)', 'sp_admm_buckets_num': 16, 'sp_admm_elem_per_row': 1, 
'sp_admm_tile': None, 'sp_admm_select_number': 4, 'sp_admm_pattern_row_sub': 1, 'sp_admm_pattern_col_sub': 4, 'sp_admm_data_format': None, 'sp_admm_do_not_permute_conv': False, 'sp_gs_output_v': 
None, 'sp_gs_output_ptr': None, 'sp_load_frozen_weights': None, 'retrain_mask_pattern': 'weight', 'sp_update_init_method': 'weight', 'sp_mask_update_freq': 10, 'retrain_mask_sparsity': -1.0, 
'retrain_mask_seed': None, 'sp_prune_before_retrain': False, 'output_compressed_format': False, 'sp_grad_update': False, 'sp_grad_decay': 0.98, 'sp_grad_restore_threshold': -1, 
'sp_global_magnitude': False, 'sp_pre_defined_mask_dir': None, 'sp_prune_ratios': 0, 'admm_sparsity_type': 'block_punched', 'admm_block': '(8,4)', 'prune_threshold': -1.0}, 'quantization': 
{'qt_aimet': False, 'qat': True, 'fold_layers': True, 'cross_layer_equalization': False, 'bias_correction': True, 'rounding_mode': 'nearest', 'num_quant_samples': 1000, 
'num_bias_correct_samples': 1000, 'weight_bw': 8, 'act_bw': 8, 'quant_scheme': 'tf_enhanced', 'layers_to_ignore': [], 'auto_add_bias': True, 'perform_only_empirical_bias_corr': True}, 'task': 
{'specific_scenarios': 'BasicTest', 'pretrained_model_path': './checkpoint/ckpt.pth', 'state': {'stage': 0, 'cycles': 0}, 'max_searching': 10}, 'user_requirements': {'power': None, 'accuracy': 
None, 'accuracy_reverse_yn': 0, 'model_size': None, 'memory_size': None, 'latency': 0.1, 'margin': 0.1, 'primary_type': 'latency', 'primary_range': '>', 'secondary_type': 'accuracy', 
'secondary_range': '<', 'searching_variable': 'scaling_factor', 'searching_range': [1, 23], 'searching_step_size': 1, 'target_type': 'latency'}, 'train': {'common_save_best_yn': 1, 'trained_yn': 
False, 'larger_better': True}, 'compiler': {'input_shape': '(1, 3, 32, 32)', 'opset_version': 11, 'devices': ['RF8MA1SEARB']}, 'distillation': {'distillation_method': None, 'enable_ddp': False, 
'enable_dp': False, 'input_shape': None, 'original_loss_weights': 0.1, 'tag_loss_weights': 0.9, 'tag_loss': 'kl', 'tag_temperature': 4, 'tag_loss_combination_method': 'avg', 
'feature_loss_weights': 0.9, 'feature_default_temperature': 1, 'advance_feature_mapping': {}, 'regularization_loss_weights': 1, 'regularization_loss_types': [], 'discriminator_lr': 0.0001}}
Current search has 1 stages
Stage:  1
Max search cycles:  1
****************************config summary************************************************
Current state: stage: 1/1| cycles: 1/1
Total jobs:1 
processing job 1/1
Training...
MKL_THREADING_LAYER=GNU CUDA_DEVICE_ORDER=PCI_BUS_ID CUDA_VISIBLE_DEVICES=1 python train_script_main.py
2022-10-01 03:23:06,951 - root - INFO - AIMET
==> Preparing data..
==> Building model..
load model from ./checkpoint/ckpt.pth
original:
testing top-1 accuracy is 95.47
2022-10-01 03:23:35,398 - CoLib - INFO - Start CoLib.init
2022-10-01 03:23:35,400 - PruneOptimizer - INFO - Start PruneOptimizer.init
2022-10-01 03:23:35,400 - QuantizationOptimizer - INFO - Start QuantizationOptimizer.init
2022-10-01 03:23:35,401 - CoLib - INFO - available compression algrithm is {'PruneOptimizer': ['Admm', 'Magnitude'], 'QuantizationOptimizer': ['AimetQt']} 
2022-10-01 03:23:35,401 - CoLib - INFO - If you find some algorithm was not shown here, please check whether fully installed the co_lib or include all co_lib related package 
2022-10-01 03:23:35,401 - CoLib - INFO - Applied algorithm is {} 
after harden:
testing top-1 accuracy is 95.47
testing top-1 accuracy is 95.47
[W NNPACK.cpp:79] Could not initialize NNPACK! Reason: Unsupported hardware.
Compiling...
Onnx file path: /tmp/8b7f7da48acd4ed18a60d8d4ab3744c5/4c3ce2ae-9b6f-41.onnx
INFO: running model, _4c3ce2ae_9b6f_41, on device RF8MA1SEARB
[W init.h:137] Caffe2 GlobalInit should be run before any other API calls.
2022-10-01 03:23:45,170 - onnx_optimizer.base.onnx_model - INFO - Sort graphs in topological order
2022-10-01 03:23:45,171 - onnx_optimizer.base.onnx_model - INFO - Output model to /tmp/tmp99i1h8ee
2022-10-01 03:23:45,632 - onnx_optimizer.base.onnx_model - INFO - Sort graphs in topological order
2022-10-01 03:23:45,633 - onnx_optimizer.base.onnx_model - INFO - Output model to /tmp/tmp99i1h8ee
2022-10-01 03:23:45,810 - onnx_optimizer.base.onnx_model - INFO - Sort graphs in topological order
2022-10-01 03:23:45,811 - onnx_optimizer.base.onnx_model - INFO - Output model to /tmp/tmpyr20_2gw
ERROR: test on device "RF8MA1SEARB" failed
The license for XGen inference engine has expired.
(xgen) root@eb2-3226-lin07:~#

**Comment by xipengshen** [comment]
@xinzhang-cocopie Can you take a look? It could be a problem of the compiler side if the "XGen inference engine" expiration message came from there.

**Comment by xinzhang-cocopie** [comment]
It looks like compiler's license expired,  I seem to remember liang had fixed a date in compiler as a validation?
If it was XGen license expired, XGen will be stoped at start.
